### ISSUE TITLE
Exclude `index.js` from the `filename-case` rule

### ISSUE THREAD

**Issue by tiagocpontesp** [sindresorhus/eslint-plugin-unicorn#131]
e.g. for `index.js` files.

**Comment by sindresorhus** [comment]
We can just exclude `index.js` by default as it has special meaning. Is there then any need for a whitelist option?

**Comment by tiagocpontesp** [comment]
I'd be happy with just that exclusion :)

**Comment by sindresorhus** [comment]
Would you be interested in doing a pull request?

**Comment by SamVerschueren** [comment]
It's probably me, but why do we need an exclusion for `index.js`? :p

**Comment by sindresorhus** [comment]
For the PascalCase option I assume.

**Comment by tiagocpontesp** [comment]
I'll look into this.. eventually.

**Comment by JavierPons** [comment]
Hi there,
Do you need collaborator?  (Im a beginner). Im looking for participate in some "easy" issue :p 
What concretely do you need?

**Comment by johnwu93** [comment]
Hi everyone,

Can I take this task? I think I can write a pull request for this.

**Comment by SamVerschueren** [comment]
Go ahead.

**Comment by johnwu93** [comment]
I'm trying do this task and I am having trouble finding a coding style for this code base. Which coding style do you guys follow?

**Comment by johnwu93** [comment]
Should we include in the schema an option to exclude `index.js` from the `filename-case` rule? Specifically, we could add the option `allowIndex` which will be a boolean. If it is true, `index.js` will be excluded from the `filename-case` rule. Otherwise, `index.js` will be taken into account for the `filename-case` rule.

The schema would look something like this:

```javascript
const schema = [{
	type: 'object',
	properties: {
		case: {
			enum: [
				'camelCase',
				'snakeCase',
				'kebabCase',
				'pascalCase'
			]
		},
		allowIndex: {
			type: 'boolean'
		}
	}
}];
```

**Comment by johnwu93** [comment]
I just finished the implementation without the option to `allowIndex`. So, all `index.js` files will be excluded from the `filename-case` rule. Please let me know if we would need the `allowIndex` option.

**Comment by tiagocpontesp** [comment]
TBH I encontered use-cases for white-listing besides "index.js". 
Namely prefixes, which come handy as a separation of some files in a dir.

I ended up disabling the rule entirely. 
Ideally I'd use a regex to target part of each filename, so not merely white-listing.
